test/tests/val-2g/val-2g.i

```diff
@@ -0,0 +1,321 @@
+    subdomain_id = '0 0 0 0 0 0 0 0 0'
+  []
+[]
+
+[Problem]
+  type = ReferenceResidualProblem
+  extra_tag_vectors = 'ref'
+  reference_vector = 'ref'
+[]
+
+[Bounds]
+  [concentration_lower_bound]
+    type = ConstantBounds
+    variable = bounds_dummy
+    bounded_variable = concentration
+    bound_type = lower
+    bound_value = ${bound_value_min}
+  []
+  [trapped_1_lower_bound]
+    type = ConstantBounds
+    variable = bounds_dummy
+    bounded_variable = trapped_1
+    bound_type = lower
+    bound_value = ${bound_value_min}
+  []
+[]
+
+[Variables]
+  [concentration]
+    order = FIRST
+    family = LAGRANGE
+    initial_condition = ${initial_concentration}
+  []
+  [trapped_1]
+    order = FIRST
+    family = LAGRANGE
+    outputs = none
+  []
+[]
+
+[AuxVariables]
+  [temperature]
+  []
+  [source]
+  []
+  [bounds_dummy]
+    order = FIRST
+    family = LAGRANGE
+  []
+[]
+
+[AuxKernels]
+  [temperature_aux]
+    type = FunctionAux
+    variable = temperature
+    function = temperature_function
+    execute_on = 'initial timestep_end linear'
+  []
+  [source_aux]
+    type = FunctionAux
+    variable = source
+    function = concentration_source_normal_function
+    execute_on = 'initial timestep_end linear'
+  []
+[]
+
+[Kernels]
+  [time_diffusion_implantation]
+    type = ADTimeDerivative
+    variable = concentration
+    extra_vector_tags = ref
+  []
+  [diffusion_implantation]
+    type = ADMatDiffusion
+    variable = concentration
+    diffusivity = diffusivity
+    extra_vector_tags = ref
+  []
+  [source]
+    type = ADBodyForce
+    variable = concentration
+    function = concentration_source_normal_function
+  []
+
+  # Trapping kernel
+  [coupled_time_trap_1]
+    type = ADCoefCoupledTimeDerivative
+    variable = concentration
+    v = trapped_1
+    coef = ${trap_per_free_1}
+    extra_vector_tags = ref
+  []
+[]
+
+[NodalKernels]
+  # First traps
+  [time_1]
+    type = TimeDerivativeNodalKernel
+    variable = trapped_1
+  []
+  [trapping_1]
+    type = TrappingNodalKernel
+    variable = trapped_1
+    mobile_concentration = concentration
+    alpha_t = '${trapping_rate_prefactor}'
+    trapping_energy = '${trapping_energy}'
+    N = '${N}'
+    Ct0 = '${trapping_site_fraction_1}'
+    temperature = 'temperature'
+    trap_per_free = ${trap_per_free_1}
+    extra_vector_tags = ref
+  []
+  [release_1]
+    type = ReleasingNodalKernel
+    variable = trapped_1
+    alpha_r = '${release_rate_prefactor}'
+    detrapping_energy = '${detrapping_energy_1}'
+    temperature = 'temperature'
+  []
+[]
+
+[BCs]
+  [left]
+    type = ADDirichletBC
+    variable = concentration
+    boundary = left
+    value = 0
+  []
+  [right]
+    type = ADDirichletBC
+    variable = concentration
+    boundary = right
+    value = 0
+  []
+[]
```

Comment:
This is a reasonable assumption, but I want to highlight that this assumption was shown to significantly affect the results compared to accounting for finite recombination kinetics at surfaces in val-2f (https://mooseframework.inl.gov/TMAP8/verification_and_validation/val-2f.html). I'm not saying the same thing should be done here, but maybe it could be mentionned in the documentation of this validation case with a link to val-2f.